schemast/annotation.go

```diff
@@ -56,6 +56,62 @@ func Annotation(annot schema.Annotation) (ast.Expr, bool, error) {
 	return fn(annot)
 }
 
+func (c *Context) AppendTypeAnnotation(typeName string, annot schema.Annotation) error {
+	stmt, err := c.annotReturnStmt(typeName)
+	if err != nil {
+		return err
+	}
+	newAnnot, shouldAdd, err := Annotation(annot)
+	if err != nil {
+		return err
+	}
+	if !shouldAdd {
+		return nil
+	}
+	returned := stmt.Results[0]
+	switch r := returned.(type) {
+	case *ast.Ident:
+		if r.Name == "nil" {
+			stmt.Results = []ast.Expr{
+				newAnnotSliceWith(newAnnot),
+			}
+			return nil
+		}
+		return fmt.Errorf("schemast: unexpected ident. expected nil got %s", r.Name)
+	case *ast.CompositeLit:
+		r.Elts = append(r.Elts, newAnnot)
+		return nil
```

Comment:
This code is repeated itself in other `Append.*` methods. Consider reusing this block in all other methods (can be done in an additional PR).

Reply:
👍